cartographer/io/fake_file_writer.cc

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/io/fake_file_writer.h"
+
+#include "glog/logging.h"
+
+namespace cartographer {
+namespace io {
+
+FakeFileWriter::FakeFileWriter(const std::string& filename,
+                               std::shared_ptr<std::string> content)
+    : is_closed_(false), content_(content), filename_(filename) {
+  CHECK(content);
+}
+
+bool FakeFileWriter::Write(const char* const data, const size_t len) {
+  if (is_closed_) return false;
```

Comment:
Consistency in the code base: Always use curly braces and multiple lines for if statements. Here it might be better though to `ASSERT_FALSE(is_closed_)` to catch wrong uses of the API. Similar below.

Reply:
ASSERT_FALSE doesn't work in this context, as the marcro will try to generate code that returns void which conflicts with the method signature. I left it like it was before. The other option would be to use CHECK().